lib/devise/models/recoverable.rb

```diff
@@ -33,8 +33,8 @@ def self.required_fields(klass)
       # Update password saving the record and clearing token. Returns true if
       # the passwords are valid and the record was saved, false otherwise.
       def reset_password(new_password, new_password_confirmation)
-        self.password = new_password
-        self.password_confirmation = new_password_confirmation
+        self.password = new_password.to_s
```

Comment:
Changed to add a blank error (maybe empty error would be better?) to the record if the new_password is nil.

I didn't define a validation for the new_password_confirmation, do you think we should?